src/main/java/bob/task/TaskList.java

```diff
@@ -0,0 +1,332 @@
+     *
+     * @param input original input given by the user.
+     */
+    public String getInputDescription(String input) throws InvalidTaskException {
+        String[] separateKeyword = input.split(" ", 2); //separate the keyword from the rest of string
+        switch (separateKeyword[0]) {
+        case "todo":
+            if (separateKeyword.length == 1) {
+                throw new InvalidTaskException("OOPS!!! The description of todo cannot be empty.");
+            }
+            return separateKeyword[1];
+        case "deadline":
+            if (separateKeyword.length == 1) {
+                throw new InvalidTaskException("OOPS!!! The description of deadline cannot be empty.");
+            }
+            String[] subString1 = separateKeyword[1].split("/by");
+            if (subString1.length <= 1) {
+                throw new InvalidTaskException("Invalid use of deadline. Should be '... /by ...'.");
+            }
+            DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
+            LocalDate x = Parser.parseDate(subString1[1].trim());
+
+            if (x == null) {
+                throw new InvalidTaskException("Invalid date format provided.");
+            }
+
+            String part2 = x.format(outputFormatter);
+            return subString1[0].trim() + " (by: " + part2 + ")";
+        case "event":
+            if (separateKeyword.length == 1) {
+                throw new InvalidTaskException("OOPS!!! The description of event cannot be empty.");
+            }
+
+            String[] subString2 = separateKeyword[1].split("/from");
+            if (subString2.length <= 1) {
+                if (subString2.length == 0) {
+                    throw new InvalidTaskException("OOPS!!! The event description cannot be empty.");
+                }
+                throw new InvalidTaskException("Invalid use of event format. "
+                        + "Should be  '<description> /from <day> <start_time> /to <end_time>'");
+            }
+
+            String[] subString3 = subString2[1].split("/to");
+            if (subString3.length <= 1) {
+                throw new InvalidTaskException("Invalid use of event format. Should be '<description> "
+                        + "/from <day> <start_time> /to <end_time>'.");
+            }
+            if (subString3[0].trim().isEmpty()) {
+                throw new InvalidTaskException("OOPS!!! The start time for the event cannot be empty.");
+            }
+            if (subString3[1].trim().isEmpty()) {
+                throw new InvalidTaskException("OOPS!!! The end time for the event cannot be empty.");
+            }
+
+            return subString2[0].trim() + " (from:" + subString3[0] + " to:" + subString3[1] + ")";
+
+        default:
+            return input;
+        }
+    }
+
+    /**
+     * Returns the String representation of the task description only.
+     * @param input input by the user.
```

Comment:
I think you could leave an empty line between the Javadoc description and the parameter

Reply:
Oh yes, I should, thanks!